lib/irb/ruby-lex.rb

```diff
@@ -136,16 +136,20 @@ def set_prompt(p = nil, &block)
     :on_param_error
   ]
 
+  def self.generate_local_variables_assign_code(context: nil, local_variables: [])
```

Comment:
I feel it weird to let `generate_local_variables_assign_code` take both `context` and `local_variables` while:

1. They're never used at the same time.
2. Local variables are all retrieved from the same context, just at different timings.

Can we make it `self.generate_local_variables_assign_code(local_variables)` instead?

And we can add something like `Context#local_variables` to encapsulate `&.workspace&.binding&.local_variables`

So that:

- We can have less methods taking `context` while what they actually need are just the local variables
- But it's also easier to generate local variables from a context
- `generate_local_variables_assign_code` can have a simpler implementation

Reply:
> `self.generate_local_variables_assign_code(local_variables)` `Context#local_variables`

It looks good, I fixed it